app/models/user_callbacks.rb

```diff
@@ -5,6 +5,13 @@ def before_create(user)
     user.unsubscribe_email_token = SecureRandom.urlsafe_base64
   end
 
+  def before_save(user)
+    return unless user.will_save_change_to_times_url?
+
+    match = user.times_url&.match(%r{\A(https://discord\.com/channels/\d+/\d+)(?:/\d+)?\Z})
```

Comment:
行末のマッチはスモールゼット `\z` ですかね？改行が入る可能性があり、それを無視したいのではればラージゼット `\Z` でも良いのですが...。ユースケースに適している方を選んでいて、タイポでないということであれば全然 OK です 🙆 

```suggestion
    match = user.times_url&.match(%r{\A(https://discord\.com/channels/\d+/\d+)(?:/\d+)?\z})
```

Reply:
https://github.com/fjordllc/bootcamp/pull/3139/files#r704850760 で改行を含まないようにスモールゼット `\z` を使うのであればこちらもスモールゼット `\z` にしておくのが良いですかね。

Reply:
`\A`との対応で`\Z`を使ってしまっていました🤯
どちらも`\z`に修正しました。